Assets/Mirror/Runtime/Transport/ClientTransport.cs

```diff
@@ -0,0 +1,136 @@
+        /// </summary>
+        public Action<Exception> OnError = (error) => Debug.LogWarning("OnError called with no handler");
+
+        /// <summary>
+        /// Notify subscribers when this client disconnects from the server
+        /// <para>callback()</para>
+        /// </summary>
+        public Action OnDisconnected = () => Debug.LogWarning("OnDisconnected called with no handler");
+
+        /// <summary>
+        /// Determines if we are currently connected to the server
+        /// </summary>
+        /// <returns>True if a connection has been established to the server</returns>
+        public abstract bool Connected();
+
+        /// <summary>
+        /// Establish a connection to a server
+        /// </summary>
+        /// <param name="address">The IP address or FQDN of the server we are trying to connect to</param>
+        public abstract void Connect(string address);
+
+        /// <summary>
+        /// Establish a connection to a server
+        /// </summary>
+        /// <param name="uri">The address of the server we are trying to connect to</param>
+        public virtual void Connect(Uri uri)
+        {
+            // By default, to keep backwards compatibility, just connect to the host
+            // in the uri
+            Connect(uri.Host);
+        }
+
+        /// <summary>
+        /// Send data to the server
+        /// </summary>
+        /// <param name="channelId">The channel to use.  0 is the default channel,
+        /// but some transports might want to provide unreliable, encrypted, compressed, or any other feature
+        /// as new channels</param>
+        /// <param name="segment">The data to send to the server. Will be recycled after returning, so either use it directly or copy it internally. This allows for allocation-free sends!</param>
+        public abstract void ClientSend(int channelId, ArraySegment<byte> segment);
+
+        /// <summary>
+        /// Disconnect this client from the server
+        /// </summary>
+        public abstract void ClientDisconnect();
+
+        /// <summary>
+        /// The maximum packet size for a given channel.  Unreliable transports
+        /// usually can only deliver small packets. Reliable fragmented channels
+        /// can usually deliver large ones.
+        ///
+        /// GetMaxPacketSize needs to return a value at all times. Even if the
+        /// Transport isn't running, or isn't Available(). This is because
+        /// Fallback and Multiplex transports need to find the smallest possible
+        /// packet size at runtime.
+        /// </summary>
+        /// <param name="channelId">channel id</param>
+        /// <returns>the size in bytes that can be sent via the provided channel</returns>
+        public abstract int GetMaxPacketSize(int channelId = Channels.DefaultReliable);
+
+        /// <summary>
+        /// Shut down the transport, both as client and server
+        /// </summary>
+        public abstract void Shutdown();
```

Comment:
what is the difference between shutdown and disconnect?

Reply:
Just checked all the transports I have imported, seems they all just do the same as stop/disconnect so Shutdown can be removed from the split versions